zerver/lib/message.py

```diff
@@ -1160,7 +1160,7 @@ def get_recent_conversations_recipient_id(user_profile: UserProfile,
     get_recent_private_conversations would have used to record that
     message in its data structure.
     """
-    my_recipient_id = user_profile.id
+    my_recipient_id = user_profile.recipient_id
```

Comment:
Can you mention the commit that introduced this bug in the commit message?  And then speculate on what effects we might have seen.